[DependencyInjection] `lint:container` fails if kernel implements interface registered for autoconfiguration

### Symfony version(s) affected

5.4+

### Description

When your Kernel implements an interface that is registered for auto-configuration, the following error occurs when running `lint:container`:

```
[ERROR] Invalid definition for service "kernel": Symfony\Component\HttpKernel\Kernel::__construct()" requires 2 arguments, 0 passed.
```

This only appears to be a problem for the lint command. The kernel is auto-configured correctly and behaves as expected.

### How to reproduce

```
git clone git@github.com:kbond/lint-container-reproducer.git
cd lint-container-reproducer
composer install
bin/console lint:container # throws exception
```

### Possible Solution

#44972.

### Additional Context

Seems similar to #40868.

Commit on my reproducer that causes the failure: https://github.com/kbond/lint-container-reproducer/commit/8676d5f93bbbf0a952aef01ae327b676efb41ceb

--- a/src/Symfony/Component/DependencyInjection/Compiler/ResolveChildDefinitionsPass.php
+++ b/src/Symfony/Component/DependencyInjection/Compiler/ResolveChildDefinitionsPass.php
@@ -114,6 +114,8 @@ class ResolveChildDefinitionsPass extends AbstractRecursivePass
 
         $def->setBindings($definition->getBindings() + $parentDef->getBindings());
 
+        $def->setSynthetic($definition->isSynthetic());
+
         // overwrite with values specified in the decorator
         $changes = $definition->getChanges();
         if (isset($changes['class'])) {

--- a/src/Symfony/Component/DependencyInjection/Tests/Compiler/ResolveChildDefinitionsPassTest.php
+++ b/src/Symfony/Component/DependencyInjection/Tests/Compiler/ResolveChildDefinitionsPassTest.php
@@ -412,4 +412,21 @@ class ResolveChildDefinitionsPassTest extends TestCase
 
         $this->process($container);
     }
+
+    public function testProcessCopiesSyntheticStatus()
+    {
+        $container = new ContainerBuilder();
+
+        $container->register('parent');
+
+        $container
+            ->setDefinition('child', new ChildDefinition('parent'))
+            ->setSynthetic(true)
+        ;
+
+        $this->process($container);
+
+        $def = $container->getDefinition('child');
+        $this->assertTrue($def->isSynthetic());
+    }
 }

